src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs

```diff
@@ -1,119 +1,78 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Runtime.CompilerServices;
-using System.IO;
-using System.Collections;
-using System.Globalization;
-using System.Text;
-using System.Threading;
+using System.Collections.Generic;
+using System.Threading.Tasks;
 using Xunit;
 
-public class DirectoryInfo_GetSetTimes
+namespace System.IO.FileSystem.Tests
 {
-    private enum TimeProperty
+    public class DirectoryInfo_GetSetTimes : FileSystemTest
     {
-        CreationTime,
-        LastAccessTime,
-        LastWriteTime
-    }
-
-    [Fact]
-    public static void ConsistencyTest()
-    {
-        String fileName = Path.Combine(TestInfo.CurrentDirectory, Path.GetRandomFileName());
+        public delegate void SetTime(DirectoryInfo testDir, DateTime time);
+        public delegate DateTime GetTime(DirectoryInfo testDir);
 
-        DirectoryInfo dir = new DirectoryInfo(fileName);
-        dir.Create();
-
-        foreach (TimeProperty timeProperty in Enum.GetValues(typeof(TimeProperty)))
+        public IEnumerable<Tuple<SetTime, GetTime, DateTimeKind>> TimeFunctions()
         {
-            if (!Interop.IsWindows && timeProperty == TimeProperty.CreationTime) // roundtripping birthtime not supported on Unix
+            if (Interop.IsWindows | Interop.IsOSX)
```

Comment:
Great idea! That would be specifically helpful for Case sensitivity since those checks are all over the place.